src/main/scala/info/bethard/timenorm/formal/Readers.scala

```diff
@@ -1,12 +1,24 @@
 package info.bethard.timenorm.formal
 
-import java.time.{DayOfWeek, Month}
+import java.time.{DayOfWeek, Month, LocalDateTime}
 import java.time.temporal.{IsoFields, WeekFields, ChronoField, ChronoUnit}
 
 import info.bethard.anafora.{Properties, Data, Entity}
 import info.bethard.timenorm.field._
 
-object AnaforaReader {
+//object AnaforaReader {
+class AnaforaReader(dct: Seq[Int])(implicit data: Data) {
```

Comment:
Make AnaforaReader take an `Interval` as dct rather than a `Seq[Int]`. Move the DCT parsing code below to your parseDCT method.